elasticdl/python/common/dataset_utils.py

```diff
@@ -17,7 +17,7 @@ def __init__(self, tasks):
         def gen(self):
             for task in self._tasks:
                 for data in self._data_reader.read_records(task):
-                    if data:
+                    if data is not None:
```

Comment:
Why?  I think `if data` is better.

Reply:
I’ve seen issues like “ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()” today with some datasets. Similar to #1112. It’s safer to asset this explicitly.

Reply:
Why does `read_records` return `None`?

Reply:
There was some mistake during implementation of new data reader. I just reverted this change since it’s unnecessary for now. Will add this in the future if really needed.